poller/poller.go

```diff
@@ -34,7 +34,7 @@ func (poller *Poller) pollForChanges() error {
 // Poll creates a goroutine and keep tracking until it stops
 func (poller *Poller) Poll() {
 	go poller.jobs()
-	<-poller.quit
+	time.Sleep(2 * time.Second)
```

Comment:
Code that sleeps for arbitrary times in an attempt to be correct, is unlikely to be correct because it's susceptible to race conditions.  It might also sleep for unnecessary time. See my other comment for correctly implementing the behavior I _think_ you want.